report/sections/Evaluation/Benchmark.tex

```diff
@@ -0,0 +1,189 @@
+These environmental constants also allow easy adjustment of Test Runner implementing the Queue System without rebuilding the docker container.
+These settings include the size of the queue system and the maximum number of threads the Test Runner can use to test the submitted Haskell code. 
+It is also possible to adjust the time between sweeps and how long each Test Runner result should be stored. 
+This makes it easy to experiment with different variables for the benchmarks and stress tests.  
+The containers are orchestrated such that the benchmarks starts only if the test container exits successfully --- that is, all the tests pass successfully. 
+Thus, wrongly configured containers do not result in misleading benchmark results.
+
+\subsection{Selecting test parameters}
+To achieve a varying degree of complexity for the submission content, several Haskell exercise solutions covering different concepts from course were selected, and Hspec tests based on the description of these problems were developed. 
+The content of the code and test submitted to the backend can be seen in Appendix \ref{chap:TestCode}.
+Each of the files covers a single topic from the course, including recursive type definitions and operations on such types, list comprehension, filtering of lists, and one containing a syntactical error.
+After selecting a representative set of exercises for stress tests and benchmarks a simple operational profile was created, and from this behavioral variables were established: an expected maximum number of clients and different poll times for these clients.
+To simulate different behavior of platform clients, classes representing clients were implemented, each representing a user for a different version of the platform.
+These clients implement representations of the necessary behavioral variables, which are annotated using BenchmarkDotNet.
+During stress testing, the BenchmarkDotNet framework creates all possible combinations of the implemented behavioral variables.
+The underlying implementation of the client classes use an HTTP client.
+To evaluate stress test and benchmark results of the developed Test Runner component, we first establish a baseline which we can compare the component to. 
+To establish such baseline, we have stress tested and benchmarked a version of the Test Runner that utilize the Rocket crate but does not implement the Queue System. 
+The benchmark for the baseline utilize the same parameterization as the benchmark for the queue system, but does not rely on polling the result after processing. 
+We suspect that when the Test Runner experience stress, the single threaded approach will experience a form of starvation where all responses will take a long time to complete due to the asynchronous processing of the request, as each request will have to compete for CPU usage with all other requests.
```

Comment:
```suggestion
We suspect that when the Test Runner experiences stress, the synchronous approach will experience a form of starvation where all responses will take a long time to complete due to the asynchronous processing of the request, as each request will have to compete for CPU usage with all other requests.
```

Reply:
I think you're mixing the two together here - the single threaded/synchronous approach just handles one request at a time in the order that they were submitted, there's no "competing" for CPU usage at all.